src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,268 @@
+            / "build"
+            / "outputs"
+            / "apk"
+            / "debug"
+            / "app-debug.apk"
+        )
+
+    def distribution_path(self, app):
+        return self.binary_path(app)
+
+    @property
+    def sdk_path(self):
+        return self.dot_briefcase / "tools" / "android_sdk"
+
+    @property
+    def sdk_url(self):
+        """The Android SDK URL appropriate to the current operating system."""
+        # The URLs described by the pattern below have existed since
+        # approximately 2017, and the code they download has a built-in
+        # updater. I hope they will work for many years.
+        return "https://dl.google.com/android/repository/" + (
+            "sdk-tools-{os}-4333796.zip".format(os=self.host_os.lower())
+        )
+
+    def verify_python_version(self):
+        if self.python_version_tag != "3.7":
+            raise BriefcaseCommandError(
+                """\
+Found Python version {self.python_version_tag}. Android packaging currently
+requires Python 3.7.""".format(
+                    self=self
+                )
+            )
+
+    def verify_sdk(self):
+        # If the SDK tools don't exist or aren't executable, install the SDK.
+        # TODO: Reconsider this for Windows.
+        tools_path = self.sdk_path / "tools" / "bin"
+        tools_ok = (
+            tools_path.exists()
+            and all(
+                [
+                    S_IMODE(tool.stat().st_mode) == 0o755
+                    for tool in tools_path.glob("*")
+                ]
+            )
+            and (tools_path / "sdkmanager").exists()
+        )
+        if tools_ok:
+            return
+
+        print("Setting up Android SDK...")
+        try:
+            sdk_zip_path = self.download_url(
+                url=self.sdk_url,
+                download_path=self.dot_briefcase / "tools",
+            )
+        except requests_exceptions.ConnectionError:
+            raise NetworkFailure("download Android SDK")
+        try:
+            with ZipFile(str(sdk_zip_path)) as sdk_zip:
+                sdk_zip.extractall(path=str(self.sdk_path))
+        except BadZipFile:
+            raise BriefcaseCommandError(
+                """\
+Invalid ZIP file found at {sdk_zip_path}
+
+Partial download? Remove it, then try again.""".format(
+                    sdk_zip_path=sdk_zip_path
+                )
+            )
+        sdk_zip_path.unlink()  # Zip file no longer needed once unpacked.
+        for binpath in tools_path.glob("*"):
+            binpath.chmod(0o755)
+
+    def verify_license(self):
+        license_path = self.sdk_path / "licenses" / "android-sdk-license"
+        if license_path.exists():
+            return
+
+        print("\nPlease accept `android-sdk-license` when prompted.\n")
+        try:
+            self.subprocess.run(
+                ["./sdkmanager", "--licenses"],
+                check=True,
+                cwd=str(self.sdk_path / "tools" / "bin"),
```

Comment:
Does this swallow stdout/stderr? 

(Related - I guess if I just delete the licenses directory I can test this myself)?

Reply:
Passes it through! And yes, that is how I test it! :)

Reply:
Resolving this conversation by adding a comment about why it is the way it is.